### ISSUE TITLE
Accidental matching on 'response' event causes infinite loop/crash

### ISSUE THREAD

**Issue by SoAG** [missinglink/huntsman#12]
Hi @missinglink,

I have been using your lib for a while and been getting this error:
```
{ [RangeError: Maximum call stack size exceeded]
  domain: 
   { domain: null,
     _events: { error: [Function] },
     _maxListeners: 10,
     members: [] },
  domainThrown: true }
```

I've figured out that the reason the exception is thrown is that the `huntsman.on('response')` callback is called over and over because the string it tries the match on contains the word `response` so it matches to the response callback even though it should not.

The line that causes it is https://github.com/missinglink/regexemitter/blob/master/index.js#L127 here an example:
```
key = 'http://www.supermarket.com/i-have-a-response-for-'
event.name = 'response'
key.match.(event.name) == true
```
so it goes off and calls huntsman.on('response') callback. I think the same is true for all `exit` and other events.

A solution would be to specify the `response` event as an exact match like: `huntsman.on('^response$')`. Another solution is to convert all strings to exact matching `RegExp` here https://github.com/missinglink/regexemitter/blob/master/index.js#L19

What are your thoughts?

I'd be happy to implement one or the other or any other solution you propose.

Thanks Roland

**Comment by SoAG** [comment]
@missinglink do you have any input on this?

**Comment by missinglink** [comment]
hi @SoAG, sorry I didn't see this, I'm pretty swamped with issues from my other projects and so completely miss stuff like this. The @ mentions work great, so please do that if I don't respond.

This is one of those cases of "oh, I didn't think of that!" but when you describe it like that it seems so obvious...

You're right in saying that this issue will affect `exit` and all other `"internal"` events.

As a quick and dirty fix I agree that using anchored strings such as `'^response$'` would work but it is not immediately obvious to someone writing an extension or looking for lower level access to internal events. For that reason I would tend towards finding a more intuitive solution.

Off the top of my head 2 solutions come to mind, the first being adding a string prefix to all `"internal"` events (as per the doc below):
http://nodejs.org/api/child_process.html#child_process_child_send_message_sendhandle

The second option is probably better as it fixes the issue directly, in that string equality should be done with `===` rather than `String.match()` here: https://github.com/missinglink/regexemitter/blob/master/index.js#L127

Your thoughts?

**Comment by SoAG** [comment]
Hi @missinglink,

Thanks for getting back to me. I've implemented the solution where strings are going to be evaluated with `===`, I just tried to push to your repository but do not have permission to do so, obviously. 
Just read/saw your last edit. I think converting to exact matching `RegExp's` still leaves it open to the bug I described since if you want to keep matching strings like `'/gp/product/'` you would need to match anywhere that string appears so `i-am-the-response-` would still be matched. So I guess that leaves prefixing the internal events to solve the problem, or am I missing something? Thanks for your help
